WallsAndHoles/abstractshapebrushtool.cpp

```diff
@@ -44,27 +42,21 @@ void AbstractShapeBrushTool::drawOverlay(int endX, int endY) {
 
     QVector<QPoint> shape = getShape(endX - mStartX, endY - mStartY);
 
-    mOverlay = Array2D<QSharedPointer<MapOverlayCell>>(getTileMap()->mapSize());
+    QRegion region;
 
-    QGraphicsScene *scene = mMapView->scene();
+    for (const QPoint &p : shape)
+        region += QRect(p.x() + mStartX, p.y() + mStartY, 1, 1);
```

Comment:
If we're committing to using QRegions, I'd be willing to change the tool code to return QRegions in getShape().